jkube-kit/generator/webapp/src/main/java/org/eclipse/jkube/generator/webapp/WebAppGenerator.java

```diff
@@ -166,27 +147,65 @@ protected Map<String, String> getEnv(AppServerHandler handler) {
 
     if ("webapps-javaee".equals(defaultEnv.get("TOMCAT_WEBAPPS_DIR"))) {
       log.info(
-          "[[B]]HINT:[[B]] Tomcat webapps dir is set to `webapps-javaee` by default for retrocompatibility. If your project is already JakartaEE compliant, set `jkube.generator.webapp.env` to `TOMCAT_WEBAPPS_DIR=webapps` for a faster startup.");
+              "[[B]]HINT:[[B]] Tomcat webapps dir is set to `webapps-javaee` by default for retrocompatibility. If your project is already JakartaEE compliant, set `jkube.generator.webapp.env` to `TOMCAT_WEBAPPS_DIR=webapps` for a faster startup.");
     }
 
     return defaultEnv;
   }
 
-  public static Map<String, String> extractEnvVariables(String envConfigValue) {
-    if (StringUtils.isBlank(envConfigValue)) {
-      return Collections.emptyMap();
-    }
+  private boolean isArtifactSameAsPrevious(File sourceFile) throws IOException, NoSuchAlgorithmException {
+    Path path = sourceFile.getParentFile().toPath();
+    byte[] data = Files.readAllBytes(sourceFile.toPath());
+    byte[] hash = MessageDigest.getInstance("MD5").digest(data);
+    String checksum = new BigInteger(1, hash).toString(16);
+    String previousChecksum = getPreviousChecksum(path);
+    log.info("Checksum for %s is %s", sourceFile.getName(), checksum);
+    log.info("Previous checksum for %s is %s", sourceFile.getName(), previousChecksum);
+    boolean isSameChecksum = checksum.equals(previousChecksum);
+    log.info("Same checksum for %s: %s", sourceFile.getName(), isSameChecksum ? "yes" : "no");
+    saveChecksum(path, checksum);
+    return isSameChecksum;
+  }
 
-    return Pattern.compile("\\n\\s*").splitAsStream(envConfigValue) // "envName1=value1\n   envName2=value2"
-        .map(envNameValue -> envNameValue.split("=")) //
-        .filter(e -> e.length == 2) //
-        .collect(Collectors.toMap(e -> e[0], e -> e[1]));
+  private void saveChecksum(Path path, String checksum) {
+    String fileName = path + File.separator + CHECKSUM_FILENAME;
+    File f = new File(fileName);
+    try (ObjectOutputStream oos = new ObjectOutputStream(Files.newOutputStream(f.toPath()))) {
+      log.info("Saving checksum for %s", path);
+      oos.writeObject(checksum);
+      oos.flush();
+    } catch (IOException e) {
+      log.warn("Failed to save checksum: " + e.getMessage());
+    }
+  }
 
+  private String getPreviousChecksum(Path path) {
```

Comment:
These methods should go to some utility class

Reply:
> These methods should go to some utility class

Yes, where should the Utility class be placed ? Should I make a util package inside the generator package and put it there?